src/sagas/background-refresh.js

```diff
@@ -40,7 +42,8 @@ function* changePage (action) {
     onNavigation: true,
     shallowFetch: true,
   }))
-  yield put(Actions.startSchedulerFixedDelay())
+  const delayInSeconds = yield select(state => state.options.getIn(['global', 'updateRate'], AppConfiguration.schedulerFixedDelayInSeconds))
+  yield put(Actions.startSchedulerFixedDelay({ immediate: true, delayInSeconds }))
```

Comment:
@rszwajko - The `immediate: true` here is causing a problem.  A few lines up, the `Actions.refresh()` call effectively is the same as the `immediate: true` bit.  So `refreshData()` is getting called twice for every `changePage()` call.

Depending on how the browser runs the 2 sagas, data gets updated at different times and you can end up ok (i.e. they both fetch the same first page), or you can end up out of sync (i.e. the 2nd invoke ends before the 1st and the vmsPage count gets out of sync with the actual number of pages fetched and then the refresh function does "strange" things going forward)

Reply:
Fixed  in separate PR #1334 The current PR will need to be rebased on top of this fix.